pkg/server/server.go

```diff
@@ -2357,6 +2357,9 @@ func (s *topLevelServer) AcceptInternalClients(ctx context.Context) error {
 					log.Ops.Errorf(connCtx, "serving SQL client conn: %v", err)
 					return
 				}
+				if status.State == pgwire.PreServeReady {
+					defer status.Reserved.Clear(ctx)
+				}
```

Comment:
[nit] Might be nice to make this a method on `PreServerStatus`?

Reply:
that sounds like a good idea